### PR TITLE
API: Add `clear` function to `bitwise`

### LINKED ISSUE #49: Add `clear` function to `bitwise.rs`
To complete API bitwise, I think we need to implement two more funcion:

- `clear_bits_msb_to_idx(&self, bit_idx: u8)` // msb: most significant bit
- `clear_bits_idx_to_0(&self, bit_idx: u8)`

### PR DESCRIPTION
**Pull request by FedericoBruzzone**
Closes #49 

These function complete the bitwise file.
- The first one allow us to clean (set to 0) all bits from `msg` to `idx`;
- The second one allow us to clean (set to 0) all bits from `idx` to `0`

Signed-off-by: Federico Bruzzone <federico.bruzzone.i@gmail.com>

### CODE REVIEW

**Comment by FedericoBruzzone** [comment]
Some function (in particular set functions) on `bitwise` file uses in its body for example `let mask = 0b1 << bit_idx` and then `*self &= mask`. 
Can I change it in `*self &= (0b1 << bit_idx)`?

File: src/bitwise.rs [hunk 1]
```diff
@@ -57,13 +59,43 @@ impl Bits for u32 {
...
+        b.clear_bits_msb_to_idx(3);
+        println!("{}", b)
+    }
+
+    #[test]
+    fn test_clear_bits_idx_to_0() {
+        let mut b = 0b11111;
+        b.clear_bits_idx_to_0(3);
+        println!("{}", b)
+    }
```
Comment (guerinoni) [inline src/bitwise.rs:97]: Tests without assertion? 🤔

File: src/bitwise.rs [hunk 2]
```diff
@@ -8,6 +8,8 @@ pub trait Bits {
     fn toggle_bit(&mut self, bit_idx: u8);
     fn set_bit(&mut self, bit_idx: u8, value: bool);
     fn get_bit(self, bit_idx: u8) -> bool;
+    fn clear_bits_msb_to_idx(&mut self, bit_idx: u8); // msb: most significant bit
+    fn clear_bits_idx_to_0(&mut self, bit_idx: u8);
```
Comment (guerinoni) [inline src/bitwise.rs:12]: Where do you we need these right now?

File: src/bitwise.rs (hunk 1 above)
Comment (FedericoBruzzone) [reply src/bitwise.rs:97]: I'm sorry, I had a big problem afternoon and i didn't check well.

File: src/bitwise.rs (hunk 2 above)
Comment (FedericoBruzzone) [reply src/bitwise.rs:12]: At the moment they will not be used but for the sake of completeness I have thrown them in.

**Review by gallottino — COMMENTED** [review]
I prefer to have only one method with left_index and right_index.

**Comment by FedericoBruzzone** [comment]
Perfect, I'll merge my two functions 👍

**Review by gallottino — APPROVED** [review]

**Comment by gallottino** [comment]
> Perfect, I'll merge my two functions 👍

I'm not sure that in this way there is an improvment. I don't like that someone has forced to start from msb or lsb.
For example:


101101

I want to clear only the two centered '1'. With this method I can't. Am I missing something?

**Comment by FedericoBruzzone** [comment]
I did not understand.

I fixed everything!

**Review by gallottino — APPROVED** [review]

**Review by gallottino — CHANGES_REQUESTED** [review]
@FedericoBruzzone There is some problem in the history: you add a void merge from main to branch and your changes are not isolated from the rest. I think that for this pull request just a commit with "add clear_bits method in bitwise" must to appear. 

Anyway we don't need this method yet, so I don't know if it is worth working on it.

**Pull request closed without merging**